components/fxa-client/src/fxa_client.udl

```diff
@@ -241,13 +280,18 @@ interface FirefoxAccount {
   // user's last-seen profile information, if any. This may be useful in helping
   // the user to reconnnect to their account. If reconnecting to the same account
   // is not desired then the application should discard the persisted account state.
-  //
+  [Throws=FxaError]
   void disconnect();
```

Comment:
As mentioned in the description, `disconnect()` now throws methods like `begin_oauth_flow()` have more cases where they throw because I added checks to make sure the auth state is what we expect.  Is this a good idea?  In many cases we could just log warnings and proceed.  I'm not sure what's best.

I made the opposite choice for `cancel_oauth_flow()`.  In that method, if we're not in an oath flow it's just a no-op.  I can't exactly explain why, but it seems like cancellation methods have more reason to be infallible.